hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/error/extractors/AbstractSingleUiErrorDetailsExtractor.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui.error.extractors;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.hawkbit.ui.error.UiErrorDetails;
+
+/**
+ * Base class for single UI error details extractors.
+ */
+public abstract class AbstractSingleUiErrorDetailsExtractor implements UiErrorDetailsExtractor {
+
+    @Override
+    public List<UiErrorDetails> extractErrorDetailsFrom(final Throwable error) {
+        return findDetails(error).map(Collections::singletonList).orElseGet(Collections::emptyList);
+    }
+
+    /**
+     * Extracts single ui error details from given error.
+     * 
+     * @param error
+     *            error to extract details from
+     * @return ui error details if found
+     */
+    protected abstract Optional<UiErrorDetails> findDetails(Throwable error);
```

Comment:
Would it make sense to have a default implementation here? It could simply take the simple name of the exception class as caption and the exception message as description.

Reply:
Not really, it is just a helper class for error details extractors that should specialise themselves on particular exception. The default case (when no extractors are found for the given exception) is managed in HawkbitUIErrorHandler#showGenericErrorNotification. There we log the detailed exception and only show generic localised error notification to the user. In theory we could use the exception class name/message in notification, however it would break the user-facing localisation so I would not do that